From e2ae981e009f76ac3481854571e740133e2a92d8 Mon Sep 17 00:00:00 2001
From: Nikhil Badola <nikhil.badola@freescale.com>
Date: Tue, 9 Dec 2014 19:00:32 +0530
Subject: [PATCH 719/741] usb: gadget: Turn on cache snooping hardware in usb
 gadget

commit e38f77267a7955b794f04079b455eb6e7deae5a4 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Turn on cache snooping hardware in usb gadget for PPC 64-bit Kernel

Signed-off-by: Nikhil Badola <nikhil.badola@freescale.com>
Signed-off-by: yinbo.zhu <yinbo.zhu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/usb/gadget/udc/fsl_udc_core.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/usb/gadget/udc/fsl_udc_core.c b/drivers/usb/gadget/udc/fsl_udc_core.c
index d70445303023..650e0421f255 100644
--- a/drivers/usb/gadget/udc/fsl_udc_core.c
+++ b/drivers/usb/gadget/udc/fsl_udc_core.c
@@ -336,6 +336,12 @@ static int dr_controller_setup(struct fsl_udc *udc)
 	}
 #endif
 
+#if (defined(CONFIG_PPC32) || defined(CONFIG_PPC64)) && \
+!defined(CONFIG_NOT_COHERENT_CACHE)
+/* Turn on cache snooping hardware, since some PowerPC platforms
+ * wholly rely on hardware to deal with cache coherent.
+ */
+
 #if !defined(CONFIG_NOT_COHERENT_CACHE)
 	/* Turn on cache snooping hardware, since some PowerPC platforms
 	 * wholly rely on hardware to deal with cache coherent. */
-- 
2.17.1

